[WebGPUSwift] make Swift.String the default String

nmahendru opened the pull request:
#### d8e90aae1092180f5e2ac7b7eb2cfc86071dc5b2
<pre>
[WebGPUSwift] make Swift.String the default String
<a href="https://rdar.apple.com/144586613">rdar://144586613</a>
<a href="https://bugs.webkit.org/show_bug.cgi?id=287462">https://bugs.webkit.org/show_bug.cgi?id=287462</a>

Reviewed by Tadeu Zagallo.

In Forward.h, we have a declaration `using WTF::String` which overrides
Swift.String type. Without this change whenever you say String, it means WTF.String
which is just an unecessary brain exercise :)
The typealias declaration added should be available to all swift files
in WebGPU project and String should mean Swift.String
from now on.
WTF.String is now WTFString.
Also, now, we don&apos;t need the ifdef in WTFString.h that hides the overloaded operator&quot;&quot;().

* Source/WTF/wtf/text/WTFString.h:
* Source/WebGPU/WebGPU/CommandEncoder.h:
(WebGPU::CommandEncoder::createCommandBuffer):
* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.validateFinishError):
(WebGPU.finish(_:)):
(WebGPU.runClearEncoder(_:depthStencilAttachmentToClear:depthAttachmentToClear:stencilAttachmentToClear:depthClearValue:stencilClearValue:existingEncoder:)):
(WebGPU.errorValidatingCopyBufferToBuffer(_:sourceOffset:destination:destinationOffset:size:)):
(WebGPU.errorValidatingCopyTextureToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingCopyTextureToBuffer(_:destination:copySize:)):
(WebGPU.errorValidatingImageCopyBuffer(_:)):
(WebGPU.errorValidatingCopyBufferToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingRenderPassDescriptor(_:)):
(WebGPU.errorValidatingTimestampWrites(_:)):
(WebGPU.errorValidatingComputePassDescriptor(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: <a href="https://commits.webkit.org/290344@main">https://commits.webkit.org/290344@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33e735a2672566b70134e70cb4043885678625ab

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/89406 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/8931 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/44287 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/94393 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/40169 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/9319 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/17209 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/68892 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/26549 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/92408 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/7155 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/81180 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/49252 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/6897 "Passed tests") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/35562 "Found 2 new test failures: editing/undo/redo-reapply-edit-command-crash.html imported/w3c/web-platform-tests/css/css-grid/grid-definition/grid-auto-fill-columns-001.html (failure)") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/39274 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/82205 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/77254 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/36561 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/96221 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/88182 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/16587 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/12202 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/77760 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/16842 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/76978 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/77068 "Passed tests") | 
| | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/21488 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/119/builds/20101 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/9737 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/14082 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/16600 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/21911 "Built successfully") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/110675 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/16341 "Built successfully") | | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/26537 "Found 191 new JSC stress test failures: jsc-layout-tests.yaml/js/script-tests/function-apply-many-args.js.layout-no-llint, microbenchmarks/interpreter-wasm.js.default, microbenchmarks/memcpy-wasm-large.js.bytecode-cache, microbenchmarks/memcpy-wasm-large.js.default, microbenchmarks/memcpy-wasm-large.js.dfg-eager, microbenchmarks/memcpy-wasm-large.js.dfg-eager-no-cjit-validate, microbenchmarks/memcpy-wasm-large.js.eager-jettison-no-cjit, microbenchmarks/memcpy-wasm-large.js.mini-mode, microbenchmarks/memcpy-wasm-large.js.no-cjit-collect-continuously, microbenchmarks/memcpy-wasm-large.js.no-cjit-validate-phases ... (failure)") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/19792 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/18122 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

mwyrzykowski commented on Source/WebGPU/WebGPU/CommandEncoder.h:
```diff
@@ -73,6 +73,12 @@ class CommandEncoder : public CommandsMixin, public RefCountedAndCanMakeWeakPtr<
         return adoptRef(*new CommandEncoder(device));
     }
+#if ENABLE(WEBGPU_SWIFT)
+    inline Ref<CommandBuffer> createCommandBuffer(id<MTLCommandBuffer> commandBuffer, Device& device, id<MTLSharedEvent> sharedEvent, uint64_t sharedEventSignalValue)
```
This change seems out of place with the others, at the least I think this can be a private member function right? It is called from CommandEncoder logic in Swift

nmahendru replied:
That was because of stacked PRs. It's no longer in this PR.

tadeuzagallo commented on Source/WebGPU/WebGPU/CommandEncoder.swift:
```diff
@@ -25,7 +25,8 @@
 import Metal
 import WebGPU_Internal
-
+public typealias WTFString = String
```
could this be `public typealias WTFString = WTF.String` just for clarity, or that wouldn't work?

nmahendru replied:
That is a good suggestion. On me to make that change in the next one.

nmahendru committed:
[WebGPUSwift] make Swift.String the default String
rdar://144586613
https://bugs.webkit.org/show_bug.cgi?id=287462

Reviewed by Tadeu Zagallo.

In Forward.h, we have a declaration `using WTF::String` which overrides
Swift.String type. Without this change whenever you say String, it means WTF.String
which is just an unecessary brain exercise :)
The typealias declaration added should be available to all swift files
in WebGPU project and String should mean Swift.String
from now on.
WTF.String is now WTFString.
Also, now, we don't need the ifdef in WTFString.h that hides the overloaded operator""().

* Source/WTF/wtf/text/WTFString.h:
* Source/WebGPU/WebGPU/CommandEncoder.h:
(WebGPU::CommandEncoder::createCommandBuffer):
* Source/WebGPU/WebGPU/CommandEncoder.swift:
(WebGPU.validateFinishError):
(WebGPU.finish(_:)):
(WebGPU.runClearEncoder(_:depthStencilAttachmentToClear:depthAttachmentToClear:stencilAttachmentToClear:depthClearValue:stencilClearValue:existingEncoder:)):
(WebGPU.errorValidatingCopyBufferToBuffer(_:sourceOffset:destination:destinationOffset:size:)):
(WebGPU.errorValidatingCopyTextureToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingCopyTextureToBuffer(_:destination:copySize:)):
(WebGPU.errorValidatingImageCopyBuffer(_:)):
(WebGPU.errorValidatingCopyBufferToTexture(_:destination:copySize:)):
(WebGPU.errorValidatingRenderPassDescriptor(_:)):
(WebGPU.errorValidatingTimestampWrites(_:)):
(WebGPU.errorValidatingComputePassDescriptor(_:)):
(WebGPU.beginComputePass(_:)):

Canonical link: https://commits.webkit.org/290344@main
